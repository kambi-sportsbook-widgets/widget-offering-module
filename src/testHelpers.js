//@flow
import { XMLHttpRequestNetworkProvider, checkStatus, jsonParser } from './utils'

const URL =
  'https://eu-offering.kambicdn.org/offering/v2018/kambi/betoffer/landing?lang=en_GB&market=UK'

const networking = (url: string = URL) =>
  XMLHttpRequestNetworkProvider(url)
    .then(checkStatus)
    .then(jsonParser)

export function getEventIdFromKambi() {
  return networking().then(data => {
    const popular = data.result.filter(res => res.name === 'popular')
    if (popular.length > 0) {
      if (popular[0].events.length > 0) {
        return popular[0].events[0].event.id
      }
    }
  })
}

export function getEventIdsFromKambi() {
  return networking().then(data => {
    const popular = data.result.filter(res => res.name === 'popular')
    if (popular.length > 0) {
      if (popular[0].events.length > 0) {
        return popular[0].events.map(({ event }) => event.id)
      }
    }
  })
}

export const configValues = {
  market: 'kambi',
  locale: 'en_GB',
  currency: 'EUR',
  channelId: 1,
  streamingAllowedForPlayer: true,
  oddsFormat: 'decimal',
  apiBaseUrl: 'https://e1-api.aws.kambicdn.com/offering/{apiVersion}/',
  apiStatisticsBaseUrl: 'https://e1-api.kambi.com/statistics/api/',
  apiBaseUrls: {
    v2: 'https://e1-api.kambi.com/offering/api/v2/',
    v3: 'https://e1-api.kambi.com/offering/api/v3/',
    v2018: 'https://e1-api.aws.kambicdn.com/offering/v2018/',
  },
  device: 'desktop',
  auth: false,
  routeRoot: '',
  offering: 'kambi',
  customer: 'kambi',
}

// expected property values to be returned from getData as a minimum
// not all properties from our flow types are used but the most vital are asserted below
export const expectedEventProps = [
  'betOffers',
  'tags',
  'id',
  'name',
  'englishName',
  'start',
  'state',
  'sport',
]

export const expectedBetOfferProps = [
  'id',
  'criterion',
  'eventId',
  'tags',
  'outcomes',
]

export const expectedKambiError = {
  body: '{"error":{"message":"No bet offers found","status":404}}',
  status: 404,
  statusText: 'Not Found',
}
