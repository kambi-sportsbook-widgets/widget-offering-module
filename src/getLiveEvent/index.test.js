import { getLiveEvent } from './index'
import { getLiveEventIdFromKambi, expectedLiveEventProps } from '../testHelpers'

jest.setTimeout(30000) // need to increase default timeout for functions as we're making network requests

describe('Test suite for getEvent(eventId) offering module function', () => {
  let response = null
  let eventId = null

  beforeAll(() => {
    response = getLiveEventIdFromKambi
  })

  it(`it gets live event information for ${eventId}`, () => {
    return response().then(id => {
      eventId = id

      return getLiveEvent(id).then(data => {
        expect(data).toBeDefined()
        expectedLiveEventProps.forEach(prop =>
          expect(data).toHaveProperty(prop)
        )
      })
    })
  })

  it('it tries to get a live event with an invalid event ID', () => {
    return getLiveEvent('123123').catch(err => {
      expect(err).toMatchSnapshot()
    })
  })

  it('it tries to get a live event without passing an event id', () => {
    expect(() => getLiveEvent()).toThrowErrorMatchingSnapshot()
  })
})
