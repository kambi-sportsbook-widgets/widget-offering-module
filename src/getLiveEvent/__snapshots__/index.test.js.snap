// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for getEvent(eventId) offering module function it tries to get a live event with an invalid event ID 1`] = `
Object {
  "body": "{\\"error\\":{\\"status\\":406,\\"message\\":\\"Not acceptable format\\"}}",
  "status": 406,
  "statusText": "Not Acceptable",
}
`;

exports[`Test suite for getEvent(eventId) offering module function it tries to get a live event without passing an event id 1`] = `"An event ID is required to return event information from Kambi Offering API"`;
